File: src/main/java/ca/yyx/hu/connection/AccessoryConnection.java
package ca.yyx.hu.connection;

import android.support.annotation.NonNull;

/**
 * @author algavris
 * @date 05/11/2016.
 */

public interface AccessoryConnection {


    interface Listener
    {
        void onConnectionResult(boolean success);
    }

    boolean isSingleMessage();
    int send(byte[] buf, int length, int timeout);
    int recv(byte[] buf, int length, int timeout);
    boolean isConnected();
    void connect(@NonNull final Listener listener);
    void disconnect();
}

File: src/main/java/ca/yyx/hu/connection/AccessoryConnection.kt
package ca.yyx.hu.connection

/**
 * @author algavris
 * *
 * @date 05/11/2016.
 */

interface AccessoryConnection {

    interface Listener {
        fun onConnectionResult(success: Boolean)
    }

    val isSingleMessage: Boolean
    fun send(buf: ByteArray, length: Int, timeout: Int): Int
    fun recv(buf: ByteArray, length: Int, timeout: Int): Int
    val isConnected: Boolean
    fun connect(listener: Listener)
    fun disconnect()
}
